tensorflow/python/training/slot_creator.py

```diff
@@ -42,12 +42,13 @@
 from tensorflow.python.framework import ops
 from tensorflow.python.ops import array_ops
 from tensorflow.python.ops import variables
+from tensorflow.python.ops import variable_scope
 
 
 def _create_slot_var(primary, val, scope):
   """Helper function for creating a slot variable."""
 
-  slot = variables.Variable(val, name=scope, trainable=False)
+  slot = variable_scope.get_variable(scope, initializer=val, trainable=False)
```

Comment:
Great! I think it's a good idea.
And I hope to make a small change to your example code. 
```python
def create_slot(primary, val, name, colocate_with_primary=True):
  with variable_scope.variable_scope(None, primary.op.name):
    return _create_slot_var(primary, val, name)
```
When I use two adam optimizers with the same kind, it will create:
```
RNN/GRUCell/Gates/Linear/Matrix/Adam/m:0 (150, 200)
RNN/GRUCell/Gates/Linear/Matrix_1/Adam/v:0 (150, 200)
RNN/GRUCell/Gates/Linear/Matrix_2/Adam/m:0 (150, 200)
RNN/GRUCell/Gates/Linear/Matrix_3/Adam/v:0 (150, 200)
```
Of course, it works. But I think `Matrix_N` is not good enough for understanding.
So I hope to change like below:
```python
def create_slot(primary, val, name, colocate_with_primary=True):
  with variable_scope.variable_scope(None, primary.op.name + '/' + name)
    return _create_slot_var(primary, val, '')
```
And now it will create like below:
```
RNN/GRUCell/Gates/Linear/Matrix/Adam/m:0 (150, 200)
RNN/GRUCell/Gates/Linear/Matrix/Adam/v:0 (150, 200)
RNN/GRUCell/Gates/Linear/Matrix/Adam/m_1:0 (150, 200)
RNN/GRUCell/Gates/Linear/Matrix/Adam/v_1:0 (150, 200)
```
Maybe `Matrix/Adam/v_1` is better than `Matrix_3/Adam/v`, do you think so?